ln-gateway/src/bin/ln_gateway.rs

```diff
@@ -75,3 +78,17 @@ fn build_federation_client(
 
     Ok(Client::new(client_cfg, db, ctx))
 }
+
+/// Persist federation client cfg to [`gateway.json`] file
+fn save_federation_client_cfg(work_dir: PathBuf, client_cfg: &GatewayClientConfig) {
```

Comment:
you can use a reference here (`work_dir: &PathBuf ..`)

Reply:
actually, clippy prefers the clone over a ref param:
```
warning: writing `&PathBuf` instead of `&Path` involves a new object where a slice will do
  --> ln-gateway/src/bin/ln_gateway.rs:83:41
   |
83 | fn save_federation_client_cfg(work_dir: &PathBuf, client_cfg: &GatewayClientConfig) {
   |                                         ^^^^^^^^ help: change this to: `&Path`
   |
   = note: `#[warn(clippy::ptr_arg)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg
```